src/ucx_plugin.c

```diff
@@ -134,26 +154,8 @@ typedef struct ucx_request {
   int                 size[NCCL_NET_IB_MAX_RECVS];
 } ucx_request_t;
 
-struct ep_list {
-  struct ncclSocket *sock;
-  struct ep_list *next;
-};
-
-/**
- * Connection descriptor. Used to store all opened connections.
- */
-struct nccl_ucx_worker {
-  ucp_context_h  ctx;      /* ucp_context bounded to specific device */
-  ucp_worker_h   worker;   /* ucp worker associated with ctx */
-  int            count;    /* number of connections that uses this worker */
-  struct ep_list *eps;     /* oob conection to all endpoints that were opened on this worker */
-  ucp_tag_t      last_tag; /* tag that last created connection uses */
-
-  struct nccl_ucx_worker *next; /* teardown list */
-};
-static struct nccl_ucx_worker workers[MAX_IB_DEVS];
-
-static struct nccl_ucx_worker *worker_gc_list = NULL;
```

Comment:
@brminich, eventually removed the gc list as you had previously commented since we still use an array to hash worker per device ;)